hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/HoodieSimpleBucketIndex.java

```diff
@@ -61,14 +75,63 @@ public Map<Integer, HoodieRecordLocation> loadBucketIdToFileIdMappingForPartitio
           if (!bucketIdToFileIdMapping.containsKey(bucketId)) {
             bucketIdToFileIdMapping.put(bucketId, new HoodieRecordLocation(commitTime, fileId));
           } else {
+            // Finding the instants which conflict with the bucket id
+            Set<String> instants = findTheConflictBucketIdInPartition(hoodieTable, partition, bucketId);
+
             // Check if bucket data is valid
             throw new HoodieIOException("Find multiple files at partition path="
-                + partition + " belongs to the same bucket id = " + bucketId);
+                + partition + " belongs to the same bucket id = " + bucketId
+                + ", these instants need to rollback: " + instants.toString()
+                + ", you can use rollback_to_instant procedure to recovery");
           }
         });
     return bucketIdToFileIdMapping;
   }
 
+
+  /**
+   * Find out the conflict files in bucket partition with bucekt id
+   */
+  public HashSet<String> findTheConflictBucketIdInPartition(HoodieTable hoodieTable, String partition, int bucketId) {
+    HashSet<String> instants = new HashSet<>();
+    HoodieTableMetaClient metaClient = hoodieTable.getMetaClient();
+    Path basePath = metaClient.getBasePathV2();
+    Path partitionPath = FSUtils.getPartitionPath(basePath, partition);
+    List<HoodieInstant> pendingInstants = metaClient.getActiveTimeline().getCommitsTimeline().filterInflightsAndRequested().getInstants();
+
```

Comment:
We have `SliceViewWithLatestSlice#getLatestFileSlicesIncludingInflight` that can return the pending file slices, also can we check by listing the files in one shot?

Reply:
OK，I would change it first @danny0405